rfcs/20200525-gelu-migration.md

```diff
@@ -0,0 +1,80 @@
+# Migrate gelu activation from TensorFlow Addons to TensorFlow Core
+
+| Status      | Proposed (Waiting for Sponsor)                                                                                           |
+| :---------- | :------------------------------------------------------------------------------------------------- |
+| **RFC #**   | [252](https://github.com/tensorflow/community/pull/252) |                                       |
+| **Authors** | Tzu-Wei Sung (@WindQAQ) & Sean Morgan (@seanpmorgan) |
+| **Sponsor** | TBD   |
+| **Updated** | 2020-06-02 |
+| **Sponsorship Deadline** | 2020-07-17 (45 Days after submission)  |
+
+## Rationale for Migration
+* [Gaussian Error Linear Units (GELUs)](https://arxiv.org/pdf/1606.08415.pdf) cited 600+ times
+* Used in BERT and other influential architectures
+* Multiple approvals from TF side: 
+    * https://github.com/tensorflow/tensorflow/pull/33945#issuecomment-617832325
+    * https://github.com/tensorflow/tensorflow/issues/32783#issuecomment-537284266
+
+## Historical Information
+* Have there been signifiant issues reported to Addons that need to be adressed?
+    * Only ABI incompatibilities for the custom-op (not an issue if built along with core TF)
+* When was it implemented in Addons?
+    * C++ custom-op added **2019-08-2019 (TFA 0.5.0)**
+    * Python composite op added **2020-02-26 (TFA 0.9.0)**
+* We have [performed benchmarking of the GELU activation](https://colab.research.google.com/drive/1rLb4EuydbFg9PbhboXhCDqopcl6BmphG#scrollTo=0GL2x2S4zxW3) 
+which shows it may be beneficial to retain the custom-ops, but the maintence burden has grown 
+to much for us to continue to support it in Addons.
+* This migration is long over-due but we've struggled with finalizing the migration process.
+
+## Implementation Details
+* Link to implementation in Addons:
+    * Python: https://github.com/tensorflow/addons/blob/r0.10/tensorflow_addons/activations/gelu.py
+    * C++ : https://github.com/tensorflow/addons/blob/r0.10/tensorflow_addons/custom_ops/activations/cc/kernels/gelu_op.h
+* Does this include custom-op kernels?
+    * Yes, but currently proposing to just migrate the python composite op. This may 
+    change with discussion in the RFC.
```

Comment:
Does this mean only python implementation will be migrated to TF?
I ask this because I saw the below topic `Is it worth bringing over the custom-op kernels for CPU/GPU?` We really hope to add a core kernel for CPU, it will get benefit from fusion, mkldnn or other optimization easily.

Reply:
adding to the discussion Zantares(Teng Lu) mentions, this nwwd for core CPU kernels over and above was also requested and mentioned in the PR by pennporn here:
https://github.com/tensorflow/tensorflow/pull/33945#issuecomment-625380208